pkg/internal/controller/metrics/metrics.go

```diff
@@ -39,6 +39,13 @@ var (
 		Help: "Total number of reconciliation errors per controller",
 	}, []string{"controller"})
 
+	// TerminalReconcileErrors is a prometheus counter metrics which holds the total
+	// number of terminal errors from the Reconciler.
+	TerminalReconcileErrors = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name: "controller_runtime_terminal_reconcile_errors_total",
```

Comment:
```suggestion
		Name: "controller_runtime_reconcile_terminal_errors_total",
```
Not sure if it's better this way around. Up to you
(if yes, then I would also rename the var to ReconcileTerminalErrors)

Reply:
I think I will just leave it as is